### ISSUE TITLE
BattlePreview has fatal error due to not recognizing scored battles as abstract battles.

### ISSUE THREAD

**Issue by JasonReed-2** [beyond-all-reason/bar-lobby#245]
**Issue**
When clicking on "Multiplayer" tab, I receive this error:
![image](https://github.com/beyond-all-reason/bar-lobby/assets/64571081/dc885ccb-8e22-4151-9218-56875b36f16b)

**Impact**
Users can't open the custom battle list browser. Also: letting this error sit with the electron app open will cause a massive slowdown as it eats up all the computers memory :(

**Investigation**
Looking at this code:
![image](https://github.com/beyond-all-reason/bar-lobby/assets/64571081/16a09bcf-def3-4f8c-9b18-797832b218bb)

When debugged: props.battle.contenders is actually ComputedRefImpl object. To access the value, it should be "props.battle.contenders.value".

So the code is recognizing the battle as a Replay when it actually should recognized as an AbstractBattle and enter the if block where the ComputedRefImpl objects are handled appropriately.

So why is the battle being treated as a replay and not a battle?

Looking in `type-checkers.ts`:
![image](https://github.com/beyond-all-reason/bar-lobby/assets/64571081/9fda56ef-b174-47dd-94b7-6ae5022aec18)

So we only recognize a battle as an abstract battle if `instanceof` abstract battle. Explanation of `instanceof` from stackoverflow: https://stackoverflow.com/questions/2449254/what-is-the-instanceof-operator-in-javascript

tldr: `instanceof` checks that the object has the constructor for the given class.

So that means that an object must be instantiated using the constructor of the class [or subclass] to be recognized by `instanceof`.

Looking at `custom.vue` in `scoreBattle` function:
![image](https://github.com/beyond-all-reason/bar-lobby/assets/64571081/78442256-84d9-4393-a4b6-8cd74e34d9ff)

So while typescript recognizes this as a `scoredSpadsBattle`: `instanceof` will not because the object wasn't created used the constructor of a subclass of `AbstractBattle`.

**Solution**
Fold the scored battles data into the `SpadsBattle` class. Have the `custom.vue` component interact with the scored battles data that is stored within the `SpadsBattle`.
